test/e2e/common_test.go

```diff
@@ -844,9 +847,27 @@ func populateCache(podman *PodmanTestIntegration) {
 	fmt.Printf("-----------------------------\n")
 }
 
+func cachePath(path string) string {
```

Comment:
I'm failing to see how this will use `$PODMAN_TEST_IMAGE_CACHE_DIR`. Should it?

Reply:
I'm bit a confused.
`ImageCacheDir` is hard corded (`/tmp/podman/imagecachedir`).
https://github.com/containers/podman/blob/a702d442e98850f56bc86ee4d6adf5f493a06201/test/e2e/config.go#L16

On the other hand,  `PODMAN_TEST_IMAGE_CACHE_DIR` also exists and used.
https://github.com/containers/podman/blob/a702d442e98850f56bc86ee4d6adf5f493a06201/test/e2e/common_test.go#L344-L357

I don't know if it is necessary to separate `ImageCacheDir` and `PODMAN_TEST_IMAGE_CACHE_DIR`.

Reply:
Sorry for taking so long to respond. I think you've hit on the main issue: `$PODMAN_TEST_IMAGE_CACHE_DIR` has nothing whatsoever to do with `ImageCacheDir`. It's a confusing similarity. I would suggest leaving the existing `$PODMAN_TEST_IMAGE_CACHE_DIR` code untouched. The reason that exists is that ginkgo tests pull enormous images, and that takes half an hour on my slow network. I need to cache those locally.

The rest of your PR, though, still needs touchup. Here's what I suggest:
* leave `imageTarPath()` untouched
* Remove lines 845-849, so `cachePath()` always returns `$TMPDIR/arg` (defaulting to `/tmp`) (as you do now)
  * Bonus points if you can get rid of `""` requirement, so callers can do `cachePath()` instead of `cachePath("")`. The former is more readable.
* in `removeCache()`, _always_ remove `cachePath("imagedir")`. Right now you don't remove it if `$TMPDIR` is set, and that seems like a logic bug.
  * bonus points if you find a way to remove using `podman unshare` or some magic namespace code, because that code always always always fails rootless.

Thanks again for your patience and for your contribution.

Reply:
I checked that the cache directory is failed to remove due to a permission.
When a rootless user, remove the cache with `podman unshare rm -rf <cacheDir>`.